document/document.go

```diff
@@ -67,15 +67,78 @@ func (fb FieldBuffer) GetByField(field string) (Value, error) {
 	return Value{}, ErrFieldNotFound
 }
 
+// treatArrayValue update the value of array at the given index.
+func treatArrayValue(vlist Array, v Value, index int) (ValueBuffer, error) {
+	var buf ValueBuffer
+	err := vlist.Iterate(func(i int, va Value) error {
+		if index == i {
+			buf = buf.Append(v)
+			return nil
+		}
+		buf = buf.Append(va)
+		return nil
+	})
+
+	return buf, err
+}
+
+// SetDotNotation allow dot notation and replace value at the given index.
+func (fb *FieldBuffer) SetDotNotation(fname string, v Value) {
+	vPath := NewValuePath(fname)
+	index, strconvErr := strconv.Atoi(vPath[1])
```

Comment:
I have trouble understanding `vPath[1]`, could you explain your reasoning?

This seems to be working for `foo.4` but not for `foo.bar.4` or `foo.bar.4.baz.1.2.3`, right?

Reply:
The `vPAth[1]` is the result of simplest unique case `foo.4` but not `foo.bar.4.baz.1.2.3`. Must be change to manage the full dot notation.